src/libs/antares/study/study.cpp

```diff
@@ -840,35 +840,40 @@ bool Study::clusterRename(Cluster* cluster, ClusterName newName)
     return ret;
 }
 
-void Study::destroyAllLoadTSGeneratorData()
+void Study::ensureDataAreLoadedForAllBindingConstraints()
 {
-    areas.each([](Data::Area& area) { area.load.prepro.reset(); });
+    for (const auto& constraint: bindingConstraints)
+    {
+        if (not JIT::IsReady(constraint->RHSTimeSeries().jit))
+        {
+            constraint->forceReload(true);
+        }
+    }
 }
 
-void Study::destroyAllSolarTSGeneratorData()
+template<>
+inline void Study::destroyTSGeneratorData<TimeSeriesType::timeSeriesLoad>()
```

Comment:
Why `inline` ?

Reply:
without it i get multiple definitions. Something to do with each translation unit that include the header create its own definition of the specialization.